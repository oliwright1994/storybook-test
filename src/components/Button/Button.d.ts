import { ButtonProps } from '@chakra-ui/core'

import { ButtonStyleType } from './constants'

export interface IButton extends Pick<ButtonProps, 'leftIcon' | 'rightIcon' | 'isDisabled'> {
  onClick: (event: React.MouseEvent<HTMLButtonElement>) => void
  style?: ButtonStyleType
  size?: 'sm' | 'lg'
}
