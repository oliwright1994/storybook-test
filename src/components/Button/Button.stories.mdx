import { Meta, Story, Props } from '@storybook/addon-docs/blocks';
import { Button } from './Button';

<Meta title="Demo/Badge" component={Badge} />

# Badge

With `MDX` we write longform markdown documentation for our `Badge` component and embed Doc Blocks inline.

<Props of={Badge} />

<Story name="positive">
  <Badge status="positive">Positive</Badge>
</Story>
