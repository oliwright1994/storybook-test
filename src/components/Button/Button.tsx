import { Button as ChakraButton, ButtonProps } from '@chakra-ui/core'
import React, { FC } from 'react'

export interface IButton extends Pick<ButtonProps, 'leftIcon' | 'rightIcon' | 'isDisabled'> {
  onClick: (event: React.MouseEvent<HTMLButtonElement>) => void
  variant?: 'primary' | 'secondary' | 'danger'
  size?: 'md' | 'sm'
}

const variantMap: Record<string, string> = {
  primary: 'green',
  secondary: 'gray',
  danger: 'red',
}

const Button: FC<IButton> = ({
  onClick,
  variant = 'primary',
  size = 'md',
  isDisabled,
  leftIcon,
  rightIcon,
  children,
}) => (
  <ChakraButton
    onClick={onClick}
    variantColor={variantMap[variant || 'primary']}
    size={size}
    isDisabled={isDisabled}
    leftIcon={leftIcon}
    rightIcon={rightIcon}
    padding="100px"
  >
    {children}
  </ChakraButton>
)

export default Button
