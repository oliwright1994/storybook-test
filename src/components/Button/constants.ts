export enum ButtonVariant {
  primary = 'primary',
  secondary = 'secondary',
  danger = 'danger',
}

export enum ButtonSize {
  sm = 'sm',
  lg = 'lg',
}
