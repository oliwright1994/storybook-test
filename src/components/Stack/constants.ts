export enum StackElement {
  div = 'div',
  ol = 'ol',
  ul = 'ul',
}

export enum StackItemElement {
  div = 'div',
  li = 'li',
}
