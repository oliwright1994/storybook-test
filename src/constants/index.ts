export * from './alignment'
export * from './messageType'
export * from './space'
export * from './alertVariant'
