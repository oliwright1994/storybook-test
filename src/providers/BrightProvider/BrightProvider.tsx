/* istanbul ignore file */
import { ThemeProvider as ChakraThemeProvider } from '@chakra-ui/core'
import React, { FC } from 'react'
import { ThemeProvider } from 'styled-components'
import { Theme } from 'styled-system'

import GlobalStyles from '../../style/GlobalStyles'
import mergeTheme from '../../theme/mergeTheme'

interface IBrightProvider {
  theme: Theme
}

const BrightProvider: FC<IBrightProvider> = ({ children, theme }) => {
  const mergedTheme = mergeTheme(theme)

  return (
    <ThemeProvider theme={theme}>
      <ChakraThemeProvider theme={mergedTheme}>
        <GlobalStyles />
        {children}
      </ChakraThemeProvider>
    </ThemeProvider>
  )
}

export default BrightProvider
