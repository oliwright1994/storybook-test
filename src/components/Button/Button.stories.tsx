import { boolean, radios, select } from '@storybook/addon-knobs'
import React from 'react'

import createStory from '../../storybook/createStory'
import Button from '.'
import { ButtonSize, ButtonVariant } from './constants'

createStory('COMPONENTS | Button', [
  {
    label: 'standard',
    jsx: () => {
      const variant = select('Variant', ButtonVariant, ButtonVariant.primary)
      const size = select('Size', ButtonSize, ButtonSize.lg)
      const disabled = boolean('Disabled', false)
      const icon = radios('Icon', { none: 'no', left: 'left', right: 'right' }, 'no')
      return (
        <Button
          onClick={() => {}}
          variant={variant}
          size={size}
          isDisabled={disabled}
          leftIcon={icon === 'left' ? 'email' : undefined}
          rightIcon={icon === 'right' ? 'email' : undefined}
        >
          I am a {variant} {size.toUpperCase()} button with {icon} icon
        </Button>
      )
    },
  },
])
