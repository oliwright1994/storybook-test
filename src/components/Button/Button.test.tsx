import { render } from '@testing-library/react'
import React from 'react'

import BrightProvider from '../../providers/BrightProvider'
import defaultTheme from '../../theme/default'
import Button from '.'
import { ButtonStyleType } from './constants'

describe('<Button />', () => {
  it('should render as expected', () => {
    const { asFragment } = render(
      <BrightProvider theme={defaultTheme}>
        <Button style={ButtonStyleType.primary} onClick={() => {}} />
      </BrightProvider>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
