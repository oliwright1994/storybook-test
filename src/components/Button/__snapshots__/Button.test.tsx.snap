// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render as expected 1`] = `
<DocumentFragment>
  <button
    class="css-1or62k6"
    type="button"
  >
    I am a button
  </button>
</DocumentFragment>
`;
